### DISCUSSION TITLE
Remote access to tasmota

### DISCUSSION THREAD

**Discussion by brkysnol** [arendst/Tasmota#11110] [Q&A] [answered]
how can I access my tasmota-installed device when I'm outside ? I use the Apple home kit. I don't have an Apple tv or tablet

**Comment by ascillato** [comment, marked as answer]
Tasmota is not designed for that. You need to use something in between.

Tasmota supports telegramBots.

Other safe option is a home automation software like Home Assistant or OpenHab.

Or use a VPN.

Never do portforwarding in your router please.

**Reply by brkysnol** [reply]
yes, I can access it with a vpn. But I can't achieve this when it's attached to the apple home kit is there anything you can suggest for it ? Can I do this with home Assistant or openhub ?

**Reply by ascillato** [reply]
Yes, you can, those home automation softwares are made for that.

**Reply by barbudor** [reply]
Why wouldn't you be able to access your Tasmota device through a VPN when integrated with Homekit ?
This is totally independent
A VPN allows your mobile to connect safely to your home network just like if you were home. So you could access those devices with the same IP address, or access your MQTT broker the same as if you were home.

In addition to @ascillato suggestions, you can also connect your Tasmota device to a cloud based MQTT broker instead of running it in your home : AWS IoT or Fespi (free) or MyqttHub (free). In that case control applications like MqttIotPanel allows to control your device wherever your are. In that case you should use MQTT with TLS for obvious security reasons.

**Reply by brkysnol** [reply]
> Yes, you can, those home automation softwares are made for that.

So without an apple tv or tablet, if I use openhab, I can access it remotely ?

**Reply by barbudor** [reply]
That's 2 different things

With a VPN, without any additional software, you can still access the Tasmota Web GUI exactly the same way you can when you are at home. So you need at least a mobile phone, a tablet or a computer with the VPN client to connect to your network (which mean a **personnal** phone/tablet/computer, not a shared one like you an find on an hotel because you need the VPN software client on this phone/tablet/computer.

With a home automation software instead of using directly Tasmota Web GUI, you can build your user-friendly dashboard and automation.

Still to access your home automation from outside your home, you would be best using a VPN. Exposing your home automation to the Internet without a VPN is possible but highly not recommended security wise.
